compiler/rustc_interface/src/passes.rs

```diff
@@ -767,6 +781,39 @@ pub static DEFAULT_EXTERN_QUERY_PROVIDERS: SyncLazy<Providers> = SyncLazy::new(|
     extern_providers
 });
 
+/// Fill the query cache with values from the resolver.
+/// Must be kept in sync with `DEFAULT_QUERY_PROVIDERS`.
+fn initialize_query_cache<'tcx>(
```

Comment:
I think it would be nicer if this is done as part of `TyCtxt::create_global_ctxt`.

Reply:
I agree.